### PR TITLE
Manually filter sold events to ensure you can only see your own sales

### PR DESCRIPTION
**Pull request by felination**
### Your checklist for this pull request

- [x Make sure you are requesting to **pull a feature/bugfix branch** (right side). Don't request from master!
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

### Checklist
* [x] Have you done your changes on a separate branch. Branches MUST have descriptive names that start with either the `fix/` or `feature/` prefixes. Good examples are: `fix/signin-loop` or `feature/pr-templates`.
* [x] Have you got descriptive commit messages that would make sense if prefixed with "This commit will ..."
* [x] Will you squash when you merge this PR?

### CODE REVIEW

File: app/src/components/SoldArtworkList.tsx [hunk 1]
```diff
@@ -44,15 +44,17 @@ class SoldArtworkList extends React.Component<ContractProps, SoldArtworkListStat
       'RecordARR', options).then(function (events: any) {
       eventsFound = events.length;
```
Comment (Tooni) [inline app/src/components/SoldArtworkList.tsx]: wont this mean that eventsFound's final value after the loop will be events.length + (number of events with right returnValues.from)?

**Review by Tooni — CHANGES_REQUESTED** [review]

File: app/src/components/SoldArtworkList.tsx (hunk 1 above)
Comment (felination) [reply app/src/components/SoldArtworkList.tsx]: You are correct, will fix

File: app/src/components/SoldArtworkList.tsx [hunk 2]
```diff
@@ -42,17 +42,18 @@ class SoldArtworkList extends React.Component<ContractProps, SoldArtworkListStat
...
-      eventsFound = events.length;
       for (let i = 0; i < events.length; i++) {
-        const soldInfo: SoldInformation = {
-          tokenId: events[i].returnValues.tokenId,
-          price: events[i].returnValues.price,
-          newOwner: events[i].returnValues.to,
-        };
-        soldInfoArray.push(soldInfo);
+        if (events[i].returnValues.from === currentAccount) {
+          eventsFound += 1;
```
Comment (Tooni) [inline app/src/components/SoldArtworkList.tsx]: do you not need an eventsFound = 0 somewhere to instantiate this?

**Review by Tooni — CHANGES_REQUESTED** [review]
it wont let me highlight it, but around line 79 you ought to add key values around each list item, else, i think but im not sure, there will be loads of warnings in console about not having unique keys. 
it would just be like `<div key={soemthing_unique_about_sale}>` `</div>` wrapping each SoldArtworkItem

not a super big deal tho

File: app/src/components/SoldArtworkList.tsx (hunk 2 above)
Comment (DJRHails) [reply app/src/components/SoldArtworkList.tsx]:
```
const SoldArtworkList: React.FC<ContractProps> = ({contracts, accounts}) => {
  const [balance, setBalance] = React.useState<number>(-1);
  const [soldList, setSoldList] = React.useState<SoldInformation[]>([]);

  const loadSales: VoidFunction = () => {
    const governance = contracts.Governance;
    const currentAccount = accounts[0];
    const filter = { filter: { from: currentAccount } };
    const options = { filter, fromBlock: 0 };

    governance.getPastEvents('RecordARR', options).then((events: any[]) => {
      setSoldList(events.map((event) => {
          return {
            tokenId: event.returnValues.tokenId,
            price: event.returnValues.price,
            newOwner: event.returnValues.to,
          };
        }
      ));

      setBalance(events.length);
    }).catch(console.log);
  }

  React.useEffect(loadSales, [balance, soldList]);

  if (balance === -1) {
    return (
      <h1>No sales to display</h1>
    );
  }

  const listItems = soldList.map((sold, index) => {
    return <SoldArtworkItem
      contracts={contracts}
      accounts={accounts}
      soldFor={sold.price}
      soldTo={sold.newOwner}
      tokenId={sold.tokenId}
      key={sold.tokenId * 3 + index * 5}
    />
  });

  return (
    <CardColumns>{listItems}</CardColumns>
  );

};

```

Comment (DJRHails) [reply app/src/components/SoldArtworkList.tsx]: Stumbled into this on my branch ^ This was my fix

Comment (felination) [reply app/src/components/SoldArtworkList.tsx]: Neil no I don’t, that’s done outside of my loops, Dan what are you attempting to fix there?

Comment (felination) [reply app/src/components/SoldArtworkList.tsx]: Oh the key values? That should be a super easy fix. Dan are you sure that with that code the filter correctly only returns sales that your account has made?

Comment (DJRHails) [reply app/src/components/SoldArtworkList.tsx]: Was also talking about the way you use eventsFound that this gets rid of

**Review by Tooni — APPROVED** [review]
fine if you're sure it works

**Pull request merged**